### PR TITLE
[AIRFLOW-4858] Deprecate "Historical convenience functions" in conf

### PR DESCRIPTION
**Pull request by haoliang7**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4858
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
The first argument of zope.deprecation.deprecated should be string or string list to deprecate top level function in a module.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by haoliang7** [comment]
In conf we want to deprecate "historical convenience functions" get, getint, getboolean, etc. However the current code doesn't issue a deprecation warning if any of these function is called. 

The first argument of zope.deprecation.deprecated should be string or string list to deprecate top level functions in a module.

**Review by ashb — CHANGES_REQUESTED** [review]
Can you please also fix up the existing uses in side airflow that this now correctly warns about?

For example a non-exhaustive list:

- airflow/config_templates/airflow_local_settings.py contains many
- airflow/lineage/__init__.py
- airflow/logging_config.py

**Comment by ashb** [comment]
~Please re-target against AIRFLOW-4723.~ Nm - related but not the same issue.

**Comment by haoliang7** [comment]
> Can you please also fix up the existing uses in side airflow that this now correctly warns about?
> 
> For example a non-exhaustive list:
> 
> * airflow/config_templates/airflow_local_settings.py contains many
> * airflow/lineage/**init**.py
> * airflow/logging_config.py

Sure, working on it.

**Comment by haoliang7** [comment]
@ashb The old config methods are removed. Please review my last commit.

File: airflow/configuration.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 # show Airflow's deprecation warnings
 warnings.filterwarnings(
-    action='default', category=DeprecationWarning, module='airflow')
+    action='always', category=DeprecationWarning, module='airflow')
```
Comment (ashb) [inline airflow/configuration.py]: Why this change? And what effect does this change have?

**Comment by ashb** [comment]
The change was a little bit bigger than needed (`configuration.conf.getboolean` was fine) but making it all consistent is no bad thing.

One small comment (on the warnings filter) then this looks good to me, thanks.

(Sorry, it's conflicted again, I didn't see the ping because my inbox is currently 1k deep in unread airflow messages 😉 )

File: airflow/configuration.py (hunk 1 above)
Comment (haoliang7) [reply airflow/configuration.py]: This was for testing and accidentally added to the commit. I have changed it back. Sorry for the confusion.

**Comment by haoliang7** [comment]
> The change was a little bit bigger than needed (`configuration.conf.getboolean` was fine) but making it all consistent is no bad thing.
> 
> One small comment (on the warnings filter) then this looks good to me, thanks.
> 
> (Sorry, it's conflicted again, I didn't see the ping because my inbox is currently 1k deep in unread airflow messages 😉 )

@ashb The conflicts have been resolved. The warnings filter was a mistake that has been corrected.  Please kindly review again. Thanks.

**Review by ashb — APPROVED** [review]
